plugins/out_s3/s3.c

```diff
@@ -129,13 +138,16 @@ static int create_headers(struct flb_s3 *ctx, char *body_md5, struct flb_aws_hea
     if (ctx->content_type != NULL) {
         headers_len++;
     }
-    if (ctx->compression == FLB_AWS_COMPRESS_GZIP) {
+    if (ctx->compression == FLB_AWS_COMPRESS_GZIP && !multipart_upload) {
         headers_len++;
     }
     if (ctx->canned_acl != NULL) {
         headers_len++;
     }
-    if (body_md5 != NULL && strlen(body_md5)) {
+    if (body_md5 != NULL && strlen(body_md5) && !multipart_upload) {
```

Comment:
this option should also be enabled with uploadpart request: https://github.com/fluent/fluent-bit/blob/master/plugins/out_s3/s3_multipart.c#L619 which can be located here: https://github.com/fluent/fluent-bit/blob/master/plugins/out_s3/s3.c#L1055. I think with this change, it can only be allowed with PutObject request then so we need to change the logic

Reply:
Thank you for your comment.

In multipart upload mode, the CreateMultipartUpload operation does not require the `Content-MD5` header. On the other hand, the UploadPart operation requires the `Content-MD5` header.
Since `create_header()` is called in `create_multipart_upload()` which calls the CreateMultipartUpload operation, there is no need to set the `Content-MD5` header in this function in multipart upload mode.

In multipart upload mode, the `Content-MD5` header is set in `upload_part()`, which handles the UploadPart operation. 
https://github.com/fluent/fluent-bit/blob/3884e3083ed6c9d0b14304404ff706a3da879756/plugins/out_s3/s3_multipart.c#L628

Reply:
Thanks for the explanation